File: web/web-frontend/src/app/model/test/event/scenario-end.event.ts
import {Path} from "../../infrastructure/path/path.model";
import {RunnerEvent} from "./runner.event";
import {ExecutionStatusEnum} from "./enums/execution-status.enum";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";
import {Scenario} from "../scenario/scenario.model";
import {ObjectUtil} from "../../../utils/object.util";

export class ScenarioEndEvent implements RunnerEvent, Serializable<ScenarioEndEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.SCENARIO_END_EVENT;

    testName: string;
    testFilePath: Path;
    scenario: Scenario;
    scenarioIndex: number;
    status: ExecutionStatusEnum;
    durationMillis: number;

    deserialize(input: Object): ScenarioEndEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.testName = input["testName"];
        this.testFilePath = Path.deserialize(input["testFilePath"]);
        this.scenario = new Scenario().deserialize(input);
        this.scenarioIndex = ObjectUtil.getAsNumber(input["scenarioIndex"]);
        let statusAsString:string = input["status"];
        this.status = ExecutionStatusEnum[statusAsString];

        this.durationMillis = input["durationMillis"];
        return this;
    }

    serialize(): string {
        return undefined;
    }
}

File: old/testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/ScenarioEndEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.test.scenario.Scenario
import com.testerum_api.testerum_steps_api.test_context.ExecutionStatus
import java.time.LocalDateTime

data class ScenarioEndEvent @JsonCreator constructor(
        @JsonProperty("time")           override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")       override val eventKey: String,
        @JsonProperty("testName")       val testName: String,
        @JsonProperty("testFilePath")   val testFilePath: Path,
        @JsonProperty("scenario")       val scenario: Scenario,
        @JsonProperty("scenarioIndex")  val scenarioIndex: Int,
        @JsonProperty("status")         val status: ExecutionStatus,
        @JsonProperty("durationMillis") val durationMillis: Long
): RunnerEvent
